src/harness/testcases/grant_testcase.py

```diff
@@ -508,34 +508,39 @@ def test_WINNF_FT_S_GRA_24(self):
     The response should be 0 (NO_ERROR)"""
 
     # Load the Data
-    pal_low_frequency = 3550000000.0
-    pal_high_frequency = 3560000000.0
-    user_id = 'pal_device'
-    device_a = json.load(
-      open(os.path.join('testcases', 'testdata', 'device_a.json')))
-    device_c = json.load(
-      open(os.path.join('testcases', 'testdata', 'device_c.json')))
-
-    pal_record = json.load(
-      open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
-    ppa_record = json.load(
-      open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
-    ppa_record, pal_record = makePpaAndPalRecordsConsistent(ppa_record,
-                                                            [pal_record],
-                                                            pal_low_frequency,
-                                                            pal_high_frequency,
-                                                            user_id)
-
-    # Move the Device to a random location in PPA
-    device_a['installationParam']['latitude'], \
-    device_a['installationParam']['longitude'] = getRandomLatLongInPolygon(ppa_record)
-    device_c['installationParam']['latitude'], \
-    device_c['installationParam']['longitude'] = getRandomLatLongInPolygon(ppa_record)
-    device_a['userId'] = user_id
-    device_c['userId'] = user_id
+    registration_request = []
+    ppa_records = []
+    pal_records = []
+    pal_frequency_0 = [3550000000.0, 3560000000.0]
+    pal_frequency_1 = [3570000000.0, 3580000000.0]
+
+    for device_filename, pal_filename, ppa_filename, frequency in zip(
+      ('device_a.json', 'device_c.json'),
+      ('pal_record_0.json', 'pal_record_1.json'),
+      ('ppa_record_0.json', 'ppa_record_1.json'),
+      (pal_frequency_0, pal_frequency_1)):
+      device = json.load(
+        open(os.path.join('testcases', 'testdata', device_filename)))
+      pal_record = json.load(
+        open(os.path.join('testcases', 'testdata', pal_filename)))
+      ppa_record = json.load(
+        open(os.path.join('testcases', 'testdata', ppa_filename)))
+      ppa_record, pal_record = makePpaAndPalRecordsConsistent(ppa_record,
+                                                              [pal_record],
+                                                              frequency[0],
+                                                              frequency[1],
+                                                              device['userId'])
+
+      # Move the Device to a random location in PPA
+      device['installationParam']['latitude'], \
+      device['installationParam']['longitude'] = getRandomLatLongInPolygon(ppa_record)
+      ppa_records.append(ppa_record)
+      pal_records.append(pal_record[0])
```

Comment:
While the naming makes it look confusing, I think you actually want `pal_records = pal_records + pal_record` (so that it will work in the case when you have multiple PAL records... just in case this is one of the test cases that becomes configurable).